src/xwayland-unmanaged.c

```diff
@@ -185,7 +184,10 @@ xwayland_unmanaged_create(struct server *server,
 	struct xwayland_unmanaged *unmanaged = znew(*unmanaged);
 	unmanaged->server = server;
 	unmanaged->xwayland_surface = xsurface;
-	xsurface->data = unmanaged;
+	/*
+	 * xsurface->data is presumed to be a (struct view *) if set,
+	 * so it must be left NULL for an unmanaged surface.
+	 */
```

Comment:
Suggest adding `xsurface->data = NULL` here to ensure that the comment holds truth in the future.
This is relevant when xwayland surfaces are being converted to unmanaged ones as that keeps the existing xwayland_surface.

We are currently setting the data pointer to `NULL` in xwayland.c `handle_destroy()` but IMHO better to be safe than sorry here.

Reply:
How about `assert(!xsurface->data)` here? The setting to NULL ought to be done earlier (where it is currently) IMO.

Reply:
Fine with me as well.

Reply:
Added `assert`